src/thor/isochrone.cc

```diff
@@ -17,10 +17,22 @@ constexpr float METRIC_PADDING = 10.f;
 template <typename PrecisionT>
 std::vector<GeoPoint<PrecisionT>> OriginEdgeShape(const std::vector<GeoPoint<PrecisionT>>& pts,
                                                   double distance_along) {
+  // just the endpoint really
+  if (distance_along == 0)
```

Comment:
if you are snapped to the end of the edge we just return a segment of zero length at the end